Model alias & removing the

fzowl opened the pull request:
- removing the prefix from sentence-transformer model
 - introducing E5 as subclass of sentence-transformer
 - introducing the ModelMeta.alias field

fzowl committed:
- adding alias field to ModelMeta
 - removing the prefix from sentence-transformer model
 - introducing E5 as subclss of sentence-transformer

fzliu commented on ebr/core/meta.py:
```diff
@@ -88,6 +88,7 @@ class ModelMeta(BaseModel):
     query_instruct: str | None = None
     corpus_instruct: str | None = None
     reference: str | None = None
+    alias: Optional[str] = None
```
Let's use `str | None = None` here to remain consistent.

fzliu commented on ebr/models/sentence_transformers.py:
```diff
@@ -14,15 +13,23 @@ def __init__(
+
+    @property
+    def _id(self) -> str:
+        return f"{self.model_name_prefix}__{self._model_meta._id}"
```
If I have some results in `output/`, does the addition of the new prefix cause the results to be recomputed?

fzowl replied:
@fzliu The changes ensures that the existing results will not be recomputed. Here, the retrieve task uses the model._id to create the output folder (please note, that it uses model._id, but only ModelMeta has the _id field!): https://github.com/fzliu/ebr/blob/f1b461e533773e3c6f7669158050dbc664d2ba75/ebr/retrieve.py#L46

EmbeddingModel does not have the _id field, but this code references to ModelMeta in this case: https://github.com/fzliu/ebr/blob/f1b461e533773e3c6f7669158050dbc664d2ba75/ebr/core/base/model.py#L33

So, what happens is that i created the _id field in the https://github.com/fzliu/ebr/blob/f1b461e533773e3c6f7669158050dbc664d2ba75/ebr/models/sentence_transformers.py#L9 class, which uses the prefix and the ModelMeta._id combined so that the resulting _id will be the same as it was prior this change.

I tested it and reverse tested, it does not recompute the existing results.

One question though: under the results/models.json, is it fine to have model names like this: `"model_name": "all-MiniLM-L6-v2",`

fzliu replied:
Yup, it's totally fine to have just the model name without the organization. (That's actually the intention for this change.)

fzowl committed:
- adding alias field to ModelMeta
 - removing the prefix from sentence-transformer model
 - introducing E5 as subclss of sentence-transformer

fzowl committed:
- adding alias field to ModelMeta
 - removing the prefix from sentence-transformer model
 - introducing E5 as subclss of sentence-transformer
